includes/fbbot/class-omise-fbbot-payment-handler.php

```diff
@@ -0,0 +1,253 @@
+    }
+
+    if ( strtolower( $wp->request ) == $payment_error_url ) {
+      $post->post_title = __( 'System error' );
+      $post->post_content = $this->payment_error_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_page_url || strtolower( $wp->request ) == $payment_error_url ) {
+      // Create cumtom page content
+      $post->post_author = 1;
+      $post->post_name = strtolower( $wp->request );
+      $post->guid = get_bloginfo( 'wpurl' ) . '/' . strtolower( $wp->request );
+      $post->ID = -1;
+      $post->post_type = 'page';
+      $post->post_status = 'static';
+      $post->comment_status = 'closed';
+      $post->ping_status = 'open';
+      $post->comment_count = 0;
+      $post->post_date = current_time( 'mysql' );
+      $post->post_date_gmt = current_time( 'mysql', 1 );
+      $posts = NULL;
+      $posts[] = $post;
+      
+      // make wp_query
+      $wp_query->is_page = true;
+      $wp_query->is_singular = true;
+      $wp_query->is_home = false;
+      $wp_query->is_archive = false;
+      $wp_query->is_category = false;
+      unset( $wp_query->query["error"] );
+      $wp_query->query_vars["error"] = "";
+      $wp_query->is_404 = false;
+    }
+
+    return $posts;
+	}
+
+	private function payment_page_render () {
+		global $wp_query;
+
+  	if ( ! isset( $wp_query->query_vars['product_id'] ) || ! isset( $wp_query->query_vars['messenger_id'] ) ) {
+  		return '<strong>404 Your order not found</strong>';
+  	}
+
+  	$product_id = $wp_query->query_vars['product_id'];
+  	$messenger_id = $wp_query->query_vars['messenger_id'];
+
+    $url = plugin_dir_path(dirname(__DIR__)) . 'templates/fbbot/payment-form.php';
+    ob_start();
+    include( $url );
+    return ob_get_clean();
+	}
+
+  private function payment_error_page_render () {
+    global $wp_query;
+
+    if ( ! isset( $wp_query->query_vars['error_message'] ) ) {
+      return '<strong>Woocommerce system has error. Please try again.</strong>';
+    }
+
+    $error_message = $wp_query->query_vars['error_message'];
+
+    return '<strong>' . $error_message . '</strong>';
+  }
+
+  public function process_payment_by_bot( $params ) {
+    /* rearrange step
+      1. receive post data from payment page
+      2. Create wc order
+      3. Create charge 
+      4. update order status
+    */
+
+    $omise_token = $params['omise_token'];
+    $product_id = $params['product_id'];
+    $messenger_id = $params['messenger_id'];
+
+    $settings = get_option( 'woocommerce_omise_settings', null );
+
+    $order = Omise_FBBot_WooCommerce::create_order( $product_id, $messenger_id );
+
+    $metadata = array(
+      'source' => 'woo_omise_bot',
+      'product_id' => $product_id,
+      'messenger_id' => $messenger_id,
+      'order_id' => $order->get_order_number()
+    );
+
+    $product = Omise_FBBot_WCProduct::create( $product_id );
+    $amount = 0;
+    $price_for_sale = $product->price;
+
+    if ( 'THB' === strtoupper( $product->currency ) ) {
```

Comment:
@igroomgrim now Omise-WooCommerce support for THB, JPY, SGD, IDR. This might not work anymore.

If consider this approach https://github.com/omise/omise-woocommerce/pull/58/files#r130411907
You will be able to use `Omise_Payment::format_amount_subunit()` method 😁